### PR TITLE
feat: unroll processor

### PR DESCRIPTION
**Pull request by schmikei**
### Proposed Change
Still need to write some documentation on this but the idea is to create new log entries out of arrays. The start of this is working exclusively with the body of a log entry.

Sample config

```yaml
receivers:
  filelog:
    include: [ ./test.json ]
    start_at: beginning
  filelog/2:
    include: [ ./test2.json ]
    start_at: beginning
processors:
  transform:
    log_statements:
      - context: log
        statements:
          - set(body, Split(body, "\\n"))
          # - unroll(body)
  transform/split-by-comma:
    log_statements:
      - context: log
        statements:
          - set(body, Split(body, ","))
  transform/set-attr:
    log_statements:
      - context: log
        statements:
          - set(attributes["owner"], "Keith")
  unroll:
  unroll/body:
    unroll_key: ""

exporters:
  file:
    path: ./test/output.json
  debug:
    verbosity: detailed
  googlecloud/otel-agent-dev:
    log:
        compression: gzip
        default_log_name: Keiths-Macbook-Pro.local
        resource_filters:
            - regex: .*
    metric:
        compression: gzip
    project: otel-agent-dev


service:
  pipelines:
    # logs:
    #   receivers: [filelog]
    #   processors: [transform, unroll]
    #   exporters: [debug, file, googlecloud/otel-agent-dev]
    logs/2:
      receivers: [filelog/2]
      processors: [transform/split-by-comma, unroll/body, transform/set-attr]
      exporters: [debug, file, googlecloud/otel-agent-dev]
```

This for example can take in a log file like this

```
1,2,3,4,5,6,7,8
```

And it becomes multiple log events:

image

##### Checklist
- [ ] Changes are tested
- [ ] CI has passed

### CODE REVIEW

File: processor/unrollprocessor/processor.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	}
+
+	return &unrollProcessor{
+		cfg: config,
+	}, nil
+}
+
+// ProcessLogs implements the processor interface
+func (p *unrollProcessor) ProcessLogs(_ context.Context, ld plog.Logs) (plog.Logs, error) {
+	newLogs := plog.NewLogs()
```
Comment (djaglowski) [inline processor/unrollprocessor/processor.go:28]: Why do we need to create a new one? Can't we just move log records around within each `LogRecordSlice`?

File: processor/unrollprocessor/processor.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	resourceLogs.CopyTo(newLogs.ResourceLogs())
+	return newLogs, errs
+}
+
+// siftScopeLogs will iterate through scope logs and copy over new logs from siftLogRecords
+func (p *unrollProcessor) siftScopeLogs(sl plog.ScopeLogsSlice) {
+	for i := 0; i < sl.Len(); i++ {
+		scopeLogs := sl.At(i)
+		l := scopeLogs.LogRecords()
+		p.siftLogRecords(l).CopyTo(scopeLogs.LogRecords())
```
Comment (djaglowski) [inline processor/unrollprocessor/processor.go:45]: Same - if we're not unrolling anything on the scope, we shouldn't need to copy or change it in any way. Just drill in and work on the `LogRecordSlice` it contains.

File: processor/unrollprocessor/processor.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	}
+}
+
+// siftLogRecords goes through each log record in the slice and will add new logs if there is a slice body
+// returns a log record slice with any logs that were not a slice body and new logs from the slice body
+func (p *unrollProcessor) siftLogRecords(logs plog.LogRecordSlice) plog.LogRecordSlice {
+	newLogRecords := plog.NewLogRecordSlice()
```
Comment (djaglowski) [inline processor/unrollprocessor/processor.go:52]: We shouldn't need to create a new copy of this. Just modify the existing one.

**Review by dpaasman00 — CHANGES_REQUESTED** [review]
Couple nits + what I messaged you on the side about

**Review by dpaasman00 — APPROVED** [review]
Looking good!

**Review by tharkhay4 — COMMENTED** [review]
Leaves

**Pull request merged**